### PR TITLE
Add new debug print method that shows line number where the print came from

### PR DESCRIPTION
**Pull request by chanon**
Implements #3884 and #18319 

I am currently using the method name `printdbg`. 

I would actually prefer to name the method `debug` or `trace` since it would be shorter. If that is ok then that would be great. However, I think `debug` might cause name collisions with people's scripts. `trace` would be safer. Another option for a short name might be `printd`

The output is displayed as:

```
Printed message		from res://path/to/file.gd:34 in function '_ready'
```

### CODE REVIEW

**Comment by chanon** [comment]
After trying this out in my project, I found the output is a bit hard to parse. 

Here is an example:
image

It looks messy.

I tried putting the source on a different line and I think it is easier to read, however it creates more lines.
image

**Comment by chanon** [comment]
After seeing how messy the previous output format was, I have decided to change the output format to look like this:

image

**Comment by mhilbrunner** [comment]
Please change it to `printd`. Thats short(er) and nice, both `trace` and `debug` could IMO cause collisions and don't fit as well.

**Comment by vnen** [comment]
I think `printdbg` is fine. I'd go with full `print_debug` TBH, since I don't understand the need for really short names. Completion helps you type this fast anyway (you can just type `debug` or `dbg` for `print_debug` to be a suggestion). Reading code should be more important than typing.

**Comment by akien-mga** [comment]
I'd go with `print_debug` too, we're always aiming to be as explicit as possible, and indeed auto-completion removes the "issue" of having a couple more keystrokes.

There are already `printt` and `prints` which could be used as argument in favour of `printd`, but IMO those two are badly named and are just keep as is for compatibility reasons.

**Comment by mhilbrunner** [comment]
Agreed, `print_debug` it is :)

**Comment by chanon** [comment]
Updated commit to be named `print_debug`
